client/extensions/woocommerce/state/ui/products/variations/edits-reducer.js

```diff
@@ -1,14 +1,23 @@
 /**
+ * External dependencies
+ */
+import { find, isEqual, uniqueId } from 'lodash';
+
+/**
  * Internal dependencies
  */
 import { createReducer } from 'state/utils';
 import {
 	WOOCOMMERCE_EDIT_PRODUCT_VARIATION,
+	WOOCOMMERCE_EDIT_PRODUCT_ATTRIBUTE,
 } from '../../../action-types';
-import { nextBucketIndex, getBucket } from '../../helpers';
+import { getBucket } from '../../helpers';
+import { editProductAttribute } from '../edits-reducer';
+import generateVariations from '../../../../lib/generate-variations';
 
 export default createReducer( null, {
 	[ WOOCOMMERCE_EDIT_PRODUCT_VARIATION ]: editProductVariationAction,
+	[ WOOCOMMERCE_EDIT_PRODUCT_ATTRIBUTE ]: editProductVariationsAction,
```

Comment:
The reducer updates variations, but it runs on `WOOCOMMERCE_EDIT_PRODUCT_ATTRIBUTE` since variations are regenerated when variation attributes change. See the previous comments here and in Slack scrollback We decided not to have a separate action for this.